### PR TITLE
[Keyboard] Code consistency updates for CRKBD

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Ensure LED count is consistent between different `config.h` for `RGBLIGHT_ENABLE`. This should ensure that the default and VIA keymap will build correctly for both `rev1` and `r2g`.

### Test compile
Built with `-e RGBLIGHT_ENABLE=yes`:
```
Checking file size of crkbd_rev1_default.hex                                                        [OK]
 * The firmware size is fine - 26778/28672 (93%, 1894 bytes free)

Checking file size of crkbd_rev1_via.hex                                                            [OK]
 * The firmware size is fine - 27394/28672 (95%, 1278 bytes free)

Checking file size of crkbd_r2g_default.hex                                                         [WARNINGS]
 * The firmware size is approaching the maximum - 27676/28672 (96%, 996 bytes free)

Checking file size of crkbd_r2g_via.hex                                                             [WARNINGS]
 * The firmware size is approaching the maximum - 28278/28672 (98%, 394 bytes free)
```

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by filterpaper** [comment]
@daskygit Here are the proposed changes to `config.h`.

**Comment by filterpaper** [comment]
@drashna RGB matrix is well supported with the current split transport code. Are there any downside if CRKBD is switched over to RGB matrix by default instead of RGB light?

@e11i0t23 Do you have any preference for `crkbd/r2g` default lighting system?

**Comment by drashna** [comment]
> @drashna RGB matrix is well supported with the current split transport code. Are there any downside if CRKBD is switched over to RGB matrix by default instead of RGB light?

From a technical standpoint, it should be fine.  but a lot of people don't use the rgb matrix feature.  But ... I'd say enable rgb matrix by default.

File: keyboards/crkbd/keymaps/default/config.h [hunk 1]
```diff
@@ -28,19 +28,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // #define MASTER_RIGHT
 // #define EE_HANDS
 
-#define USE_SERIAL_PD2
-
-#define TAPPING_FORCE_HOLD
-#define TAPPING_TERM 100
+//#define USE_SERIAL_PD2
```
Comment (drashna) [inline keyboards/crkbd/keymaps/default/config.h:31]: This should be removed, outright. 
```suggestion
```

File: keyboards/crkbd/keymaps/default/config.h [hunk 2]
```diff
@@ -28,19 +28,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+//#define TAPPING_FORCE_HOLD
+//#define TAPPING_TERM 100
 
 #ifdef RGBLIGHT_ENABLE
-    #undef RGBLED_NUM
     #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/default/config.h:36]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/crkbd/keymaps/via/config.h [hunk 3]
```diff
@@ -25,16 +25,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#    define RGBLIGHT_ANIMATIONS
-#    define RGBLED_NUM 54
-#    undef RGBLED_SPLIT
-#    define RGBLED_SPLIT \
-        { 27, 27 }
-#    define RGBLIGHT_LIMIT_VAL 120
-#    define RGBLIGHT_HUE_STEP  10
-#    define RGBLIGHT_SAT_STEP  17
-#    define RGBLIGHT_VAL_STEP  17
+    #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/via/config.h:28]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
// #define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by filterpaper** [comment]
> From a technical standpoint, it should be fine. but a lot of people don't use the rgb matrix feature. But ... I'd say enable rgb matrix by default.

On second thoughts, it might be good to leave RGB light as default for `default` and `via`. You've provided clear instructions on the top level readme for enabling RGB matrix.

File: keyboards/crkbd/keymaps/default/rules.mk [hunk 4]
```diff
@@ -1,2 +1,5 @@
-OLED_ENABLE = yes
-OLED_DRIVER = SSD1306
+MOUSEKEY_ENABLE = no     # Mouse keys
```
Comment (daskygit) [inline keyboards/crkbd/keymaps/default/rules.mk:1]: What's the reasoning behind disabling mouse key support? Is it just that the default keymap doesn't use them?

My concern here being the new-keymap command copies these rules and the user might be used to mouse keys working from configurator. Although admittedly it is a easy fix but it's still a often asked question.

**Review by daskygit — COMMENTED** [review]
This doesn't put any of the existing keymaps oversize that aren't already broken.

I don't know if this could be classed as a bug fix and target master, as the `#define RGBLED_NUM 27` in particular breaks RGBLIGHT functionality, encountered this a lot with new users making use of the `new-keymap` command for the r2g revision.

Maybe drashna has some thoughts on this?

File: keyboards/crkbd/keymaps/default/rules.mk (hunk 4 above)
Comment (filterpaper) [reply keyboards/crkbd/keymaps/default/rules.mk:1]: Thanks for pointing it out. It was an error on my part from copying the `rules.mk` over from `via` keymap. I've reduced the number of effects for the `via` keymap as well to fit mouse key.

**Comment by filterpaper** [comment]
> This doesn't put any of the existing keymaps oversize that aren't already broken.
> 
> I don't know if this could be classed as a bug fix and target master, as the `#define RGBLED_NUM 27` in particular breaks RGBLIGHT functionality, encountered this a lot with new users making use of the `new-keymap` command for the r2g revision.
> 
> Maybe drashna has some thoughts on this?

Rebased to target master.

**Review by daskygit — CHANGES_REQUESTED** [review]
This currently compiles over size for 5.4. (via keymap)

>The firmware is too large! 28704/28672 (32 bytes over)

**Review by daskygit — APPROVED** [review]
I'm happy to class this as a bug fix but I'll defer to someone with better judgement.

**Review by drashna — APPROVED** [review]

**Comment by filterpaper** [comment]
Thank you

**Pull request merged**